core/src/main/java/org/springframework/security/authentication/AuthenticationTrustResolverImpl.java

```diff
@@ -52,6 +54,10 @@ public boolean isAnonymous(Authentication authentication) {
 			return false;
 		}
 
+		if (mfaTokenEvaluator != null && mfaTokenEvaluator.isMultiFactorAuthentication(authentication)) {
```

Comment:
The user is not anonymous if the token is a multi factor authentication. This implementation should not consult `isMultiFactorAuthentication` to determine if the user is authenticated

Reply:
`MultiFactorAuthenticationToken` represents a principal in the middle of multi factor (stage) authentication.
In the case of two-step authentication using a password and a security key, it indicates that the password has already been provided and authentication using the security key has not been completed.
Since authentication has not yet finished fully, I think it is appropriate to treat it as anonymous.
Do you think the name `MultiFactorAuthenticationToken` is inappropriate? I want a suggestion for a good name.